Update primary email if unconfirmed access allowed

swrobel opened the pull request:
It makes sense to me that confirmable should allow setting primary to true for a new record when allow_unconfirmed_access_for is set to greater than the default 0.days

swrobel commented:
@allenwq just a quick ping, I fixed the tests so everything's passing

allenwq commented:
@swrobel Thanks, make sense!

joelvh commented:
hi @swrobel why did you change the existing test? Will this change the defaults for everyone? It's unclear from the change what the impact will be for previous versions of this gem.

Thanks.

allenwq commented:
@joelvh right, think about it again, it worths a discussion.  

Our old behaviour is: `allow_unconfirmed` is only apply to the primary email and users are not able to change a non-primary unconfirmed email to primary. 

This PR allows that.  I think this might break the existing app. i.e. if the primary email is already confirmed, but in this case we allow adding a new unconfirmed primary email... which doesn't make sense to me. 

@swrobel what's your use case for this scenario?

allenwq commented:
@swrobel  I see, but your change breaks one test which allows non-primary unconfirmed email to sign in. any ideas?

swrobel commented:
@allenwq that is expected, since I changed the behavior in this scenario. There are 3 possible scenarios for `email=` which are all tested:

1. New email is confirmed at creation time. This changes the user's primary email to the new email.
2. New email is unconfirmed at creation time and unconfirmed access is *disallowed* via config. This does not change the user's primary email to the new email.
3. New email is unconfirmed at creation time but unconfirmed access is *allowed* via config. This is the scenario in question which I changed the behavior & test for. Previously, the behavior was the same as 2, but I changed it to be the same as 1. Certainly, the test could be rewritten to better explain what is happening, since the new email is not a "non-primary" email, but is in fact now the primary email. I can submit a PR with a change to that test description, as well as an additional test case for [Change primary email](https://github.com/swrobel/devise-multi_email/blob/f6b51429a84bacad8f7c953e319abf36c3fecaad/spec/features/confirmable_spec.rb#L39), as long as we all agree that this behavior in scenario 3 is correct.

allenwq commented:
> since the new email is not a "non-primary" email, but is in fact now the primary email.

I see. How about we revert the test, and update `create_email` to not assign the secondary email to primary instead? or use a new method to create secondary emails with out using `email=`

allenwq commented:
>  an additional test case for Change primary email, as long as we all agree that this behavior in scenario 3 is correct.

this sounds good! in fact scenario 3 didn't change the behaviour  so I am okay with that. it's rather the test helpers are confusing. 

@swrobel
